cumulusci/tasks/marketing_cloud/deploy.py

```diff
@@ -92,8 +100,15 @@ def _construct_payload(self, dir_path):
                 with open(item, "r") as f:
                     payload["entities"][entity_name][entity_id] = json.load(f)
 
+        if custom_inputs:
+            self._add_custom_inputs_to_payload(custom_inputs, payload)
+
         return payload
 
+    def _add_custom_inputs_to_payload(self, custom_inputs, payload):
+        for input_name, value in custom_inputs.items():
+            payload["input"].append({"key": input_name, "value": value})
```

Comment:
My understanding is that the package exported from MCPM will already have `input` set to a list of the supported inputs for the package. So for each key, we need to find the matching entry in that list and add the value to it, not append a new item to the list.

This also means that we can throw an error if an input key is specified that is not supported by the package.